payments-core/src/test/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerTest.kt

```diff
@@ -326,16 +330,19 @@ internal class DefaultFlowControllerTest {
         // up the payment option launcher
         flowController.onPaymentOptionResult(PaymentOptionResult.Succeeded(SAVE_NEW_CARD_SELECTION))
 
-        // Save off the actual launch arguments when paymentOptionLauncher is called
-        var launchArgs: PaymentOptionContract.Args? = null
-        flowController.paymentOptionLauncher = {
-            launchArgs = it
-        }
+        // capture the actual launch arguments when paymentOptionLauncher is called
+        val mockPaymentOptionActivityLauncher =
+            mock<ActivityResultLauncher<PaymentOptionContract.Args>>()
+        val paymentOptionsContractArgsCaptor = argumentCaptor<PaymentOptionContract.Args>()
+
+        flowController.paymentOptionActivityLauncher = mockPaymentOptionActivityLauncher
 
         flowController.presentPaymentOptions()
 
+        verify(mockPaymentOptionActivityLauncher).launch(paymentOptionsContractArgsCaptor.capture())
+
         // Make sure that paymentMethods contains the new added payment methods and the initial payment methods.
-        assertThat(launchArgs!!.paymentMethods)
+        assertThat(paymentOptionsContractArgsCaptor.firstValue.paymentMethods)
             .isEqualTo(initialPaymentMethods)
```

Comment:
Instead of using an arg captor, I like using `argWhere`

```
verify(mockPaymentOptionActivityLauncher).launch(
  argWhere {
    it.paymentMethods == initialPaymentMethods
  }
)
```

Reply:
neat!